opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -361,6 +361,14 @@ def __init__(
         self.metrics_lock = threading.Lock()
         self.observers_lock = threading.Lock()
         self.view_manager = ViewManager()
+        self.instrument_names = set()
+
+    def _check_instrument_name(self, name: str):
+        if name in self.instrument_names:
+            raise ValueError(
+                "Multiple instruments can't registered by the same name"
```

Comment:
Would be nice to also include the name in the exception

```suggestion
                "Multiple instruments can't be registered by the same name"
```